### ISSUE TITLE
SVG usage in blockreassurance is broken

### ISSUE THREAD

**Issue by matks** [PrestaShop/PrestaShop#25458]
#### Describe the bug

Blockreassurance blocks crash on FO when using an SVG icon and `__PS_BASE_URI__` is a slash

#### Expected behavior

No crash, correct behavior and display of blockreassurrance blocks on FO

#### Steps to Reproduce

1. You use blockreassurance v5.0.0 and `__PS_BASE_URI__` is a single slash `/` (default situation)
2. Click on 'configure' for blockreassurance module, edit one block
3. Upload a custom icon: use a SVG file
4. Open FO product page, see error `Warning: finfo_file(...): failed to open stream: No such file or directory`

**Screenshots**

![Capture d’écran 2021-07-29 à 09 32 54](https://user-images.githubusercontent.com/3830050/127450756-42eece5e-09ca-4ba2-8539-3f88fcc09022.png)

As mentioned by @matthieumota in PR https://github.com/PrestaShop/blockreassurance/pull/210 when `__PS_BASE_URI__` is a `/` the `str_replace()` replace all slashes by project directory. This is the source of the bug.

#### Additional information

* PrestaShop version: 1.7.8.x
* PHP version: 7.4

**Comment by matks** [comment]
PR available https://github.com/PrestaShop/blockreassurance/pull/210

**Comment by hibatallahAouadni** [comment]
Hello @matks 

This issue reminds me of this one #15939 whose already fixed by this PR https://github.com/PrestaShop/blockreassurance/pull/35
Is it related?

Thanks!

**Comment by matks** [comment]
No, it is different. PR https://github.com/PrestaShop/blockreassurance/pull/35 enabled the use of SVG but contained a little mistake. This mistake introduced a bug, this is the bug being reported here. PR https://github.com/PrestaShop/blockreassurance/pull/210 fixes this bug, so you could say https://github.com/PrestaShop/blockreassurance/pull/210 is "part 2" of https://github.com/PrestaShop/blockreassurance/pull/35

**Comment by hibatallahAouadni** [comment]
Well then, thanks for reporting the issue :wink:

**Comment by olecorre** [comment]
I have the same problem on a new installation, I had to apply the patch

**Comment by matks** [comment]
Looks similar to https://github.com/PrestaShop/PrestaShop/issues/28264

**Comment by hibatallahAouadni** [comment]
@matks so should I close it as a duplicate?

**Comment by matks** [comment]
@hibatallahAouadni https://github.com/PrestaShop/PrestaShop/issues/28264 is being fixed so I suggest we wait for https://github.com/PrestaShop/PrestaShop/issues/28264 then we verify if this one is still present.

**Comment by boo-code** [comment]
@matks asked to re-verify once #28264 was fixed. #28264 closed 2023-01-24, so: this does not reproduce on blockreassurance 6.0.0.

Measured on a shop with `__PS_BASE_URI__` = `/` (the reported condition), an SVG uploaded as a custom icon, and the module active. The FO product page returns 200, renders `img_perso/<icon>.svg`, and contains no `failed to open stream` warning.

Neither of the two proposed PRs landed - blockreassurance#210 and #376 are both closed unmerged. What fixed it is the move to the Doctrine repository. The live path is now `PsreassuranceRepository::getAllBlockByStatus()`, which builds the path from the upload folder:

```
$static_folder_file_upload . $item['custom_icon']
  -> /var/www/html/modules/blockreassurance/views/img/img_perso/<icon>.svg   exists, mime image/svg+xml
```

`$folder_file_upload` is `_PS_MODULE_DIR_ . $this->name . '/views/img/img_perso/'`, an absolute filesystem path, so the base URI no longer takes part in it and the single-slash case cannot recur.

As a control that the check can actually detect the failure, the old expression on the same shop:

```
_PS_ROOT_DIR_ . $item['custom_icon']
  -> /var/www/htmlqa.svg   does not exist
  Warning: finfo_file(/var/www/htmlqa.svg): Failed to open stream: No such file or directory
```

which is the warning from the report.

One residual, not the reported bug but worth knowing before this is closed: `classes/ReassuranceActivity.php:169-170` on `dev` still carries that old expression. It is unreachable - `ReassuranceActivity` is referenced by no other file in the repository, the Doctrine entity `src/Entity/Psreassurance.php` having superseded it - so it cannot fire today, but it would if the class were ever called again. Deleting the class or aligning those two lines with the repository would both close that off; the first is a call on whether the class counts as public API.

Suggest closing this one.
